routes/repo/repo.go

```diff
@@ -235,7 +236,18 @@ func Action(c *context.Context) {
 	case "watch":
 		err = models.WatchRepo(c.User.ID, c.Repo.Repository.ID, true)
 	case "unwatch":
-		err = models.WatchRepo(c.User.ID, c.Repo.Repository.ID, false)
+		if userID := c.Query("user_id"); userID != "" {
```

Comment:
Sorry I forgot to mention in the last round of review, there is a better method:

https://github.com/gogs/gogs/blob/ef2e29db725c573d0d9bef37bcee56022aeea3c9/vendor/gopkg.in/macaron.v1/context.go#L240-L243

Reply:
thanks